docs/manual/visualization.md

```diff
@@ -1,7 +1,129 @@
+/vis/geometry/set/colour {volume} 0 {r} {g} {b} {a}
+# hide the volume {volume}.
+/vis/geometry/set/visibility {volume} -1 false
+```
+
+The Geant4 visualization can also color tracks by properties of the particles.
+The following example macro illustrates this in the case of optical photons,
+colored by their wavelength.
+
+````{admonition} coloring photon tracks by wavelength
+:class: dropdown
+
+```geant4
+/vis/modeling/trajectories/create/drawByAttribute
+/vis/modeling/trajectories/select drawByAttribute-0
+/vis/modeling/trajectories/drawByAttribute-0/verbose false
+/vis/modeling/trajectories/drawByAttribute-0/setAttribute IMag
+
+/vis/modeling/trajectories/drawByAttribute-0/addInterval vuviolet 6.53 eV 12.4 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval duviolet 4.43 eV 6.53 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval muviolet 3.94 eV 4.43 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval nuviolet 3.26 eV 3.94 eV
+
+/vis/modeling/trajectories/drawByAttribute-0/addInterval violet  2.85 eV 3.26 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval blue    2.48 eV 2.85 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval cyan    2.38 eV 2.48 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval green   2.19 eV 2.38 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval yellow  2.10 eV 2.19 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval orange  1.98 eV 2.10 eV
+/vis/modeling/trajectories/drawByAttribute-0/addInterval red     1.59 eV 1.98 eV
+
+/vis/modeling/trajectories/drawByAttribute-0/vuviolet/setLineColourRGBA 0.29 0.00 0.51 1
+/vis/modeling/trajectories/drawByAttribute-0/duviolet/setLineColourRGBA 0.50 0.00 0.50 1
+/vis/modeling/trajectories/drawByAttribute-0/muviolet/setLineColourRGBA 0.58 0.00 0.83 1
+/vis/modeling/trajectories/drawByAttribute-0/nuviolet/setLineColourRGBA 0.73 0.33 0.83 1
+
+/vis/modeling/trajectories/drawByAttribute-0/violet/setLineColourRGBA  0.93 0.51 0.93 1
+/vis/modeling/trajectories/drawByAttribute-0/blue/setLineColourRGBA    0.00 0.00 1.00 1
+/vis/modeling/trajectories/drawByAttribute-0/cyan/setLineColourRGBA    0.00 1.00 1.00 1
+/vis/modeling/trajectories/drawByAttribute-0/green/setLineColourRGBA   0.00 1.00 0.00 1
+/vis/modeling/trajectories/drawByAttribute-0/yellow/setLineColourRGBA  1.00 1.00 0.00 1
+/vis/modeling/trajectories/drawByAttribute-0/orange/setLineColourRGBA  1.00 0.65 0.00 1
+/vis/modeling/trajectories/drawByAttribute-0/red/setLineColourRGBA     1.00 0.00 0.00 1
+```
+
+````
+
+### Offscreen rendering with ToolsSG
+
+Using the ToolsSG offscreen rendering to render to a file is easy:
+
+```geant4
+# after /run/initialize
+/vis/open TOOLSSG_OFFSCREEN 1500x1500
+
+# your simulation commands here...
+# /run/beamOn ...
+
+/vis/tsg/offscreen/set/file {filename}.jpeg
+/vis/viewer/rebuild
+```
+
+## Interactive visualization
+
+The interactive mode `remage -i` can be used together with the visualization and
```

Comment:
can we list the useful env variables in this case? for example, i remember choosing a session type at runtime via env variable that suppressed the gui